### PR TITLE
[fix] create missing root's `__data.json`

### PR DESCRIPTION
**Pull request by baseballyama**
fix: https://github.com/sveltejs/kit/issues/3974

The cause was that root's `__data.json` had not been created.
It seems to be a SPA-specific issue, so I created a test in adapter-static.

--- 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/runtime/server/page/load_node.js [hunk 1]
```diff
@@ -346,7 +346,7 @@ export async function load_node({
 
 	// generate __data.json files when prerendering
 	if (shadow.body && state.prerender) {
-		const pathname = `${event.url.pathname}/__data.json`;
+		const pathname = `${event.url.pathname.replace(/\/$/, '')}/__data.json`;
```
Comment (Conduitry) [inline packages/kit/src/runtime/server/page/load_node.js:349]: Are there other paths besides literally `'/'` that might end with a `/` that we have to worry about here? Can we check for the literal string `'/'` here which would (presumably) be slightly less computationally intensive?

Comment (Rich-Harris) [reply packages/kit/src/runtime/server/page/load_node.js:349]: yeah, this is also fixing the `trailingSlash: 'always'` case

**Review by Rich-Harris — APPROVED** [review]

File: packages/kit/src/runtime/server/page/load_node.js (hunk 1 above)
Comment (Conduitry) [reply packages/kit/src/runtime/server/page/load_node.js:349]: Gotcha, okay, that makes me sufficiently disinterested in microoptimizations, then,

**Comment by baseballyama** [comment]
Ah yes. Thank you for simplifying the test!

**Comment by Rich-Harris** [comment]
Thank you for the fix!

**Pull request merged**